blobmodel/blobs.py

```diff
@@ -20,34 +20,46 @@ class Blob():
     t_drain: float
 
     def discretize_blob(self, x, y, t):
-        ''' 
+        '''
         Discretize blob on grid
         The following blob shapes are implemented:
                 gauss: 2D gaussian function
                 exp: one sided exponential in x and gaussian in y
         '''
         if(self.blob_shape == 'gauss'):
-            return self.amplitude * np.exp(-(t-self.t_init)/self.t_drain)\
-                * np.exp(-((x-self.pos_x - self.v_x*(t-self.t_init))**2/(2*self.width_x**2)))\
-                * np.exp(-((y-self.pos_y - self.v_y*(t-self.t_init))**2/(2*self.width_y**2)))\
-                * np.heaviside(t-self.t_init, 1)
+            return self.amplitude\
+                * self.__drain(t)\
+                * self.__x_shape_gauss(x, t)\
+                * self.__y_shape_gauss(y, t)\
+                * self.__blob_arrival(t)
         elif(self.blob_shape == 'exp'):
-            return self.amplitude * np.exp(-(t-self.t_init)/self.t_drain)\
-                * np.exp(x - self.pos_x - self.v_x*(t-self.t_init))\
-                * np.exp(-((y-self.pos_y - self.v_y*(t-self.t_init))**2/(2*self.width_y**2)))\
-                * np.heaviside(t-self.t_init, 1)*np.heaviside(-1.*(x - self.pos_x - self.v_x*(t-self.t_init)), 1)
+            return self.amplitude\
+                * self.__drain(t)\
+                * self.__x_shape_exp(x, t)\
+                * self.__y_shape_gauss(y, t)\
+                * self.__blob_arrival(t)
```

Comment:
I think there should be no if-else statement here. Instead only something like 
            return self.amplitude\
                * self.__drain(t)\
                * self.__x_shape(x, t)\
                * self.__y_shape(y, t)\
                * self.__blob_arrival(t)
              
And then the x_shape and y_shape functions will have the if-else statement. This way you reduce code duplication

Reply:
I changed the code accordingly, what do you think now?